ThingIFSDK/ThingIFSDK/ThingIFAPI.swift

```diff
@@ -366,23 +366,32 @@ public class ThingIFAPI: NSObject, NSCoding {
 
     **Note**: Please onboard first, or provide a target instance by calling copyWithTarget. Otherwise, KiiCloudError.TARGET_NOT_AVAILABLE will be return in completionHandler callback
 
+    When thing related to this ThingIFAPI instance meets condition
+    described by predicate, A registered command sends to thing
+    related to target.
+
+    `target` property and target argument must be same owner's things.
+
     - Parameter schemaName: Name of the Schema of which the Command specified in
     Trigger is defined.
     - Parameter schemaVersion: Version of the Schema of which the Command
     specified in Trigger is defined.
     - Parameter actions: Actions to be executed by the Trigger.
     - Parameter predicate: Predicate of the Command.
+    - Parameter target: target of trigger. This is optional. If you
+      ommit or set nil to this parameter, `target` property is used.
     - Parameter completionHandler: A closure to be executed once finished. The closure takes 2 arguments: 1st one is an created Trigger instance, 2nd one is an ThingIFError instance when failed.
     */
     public func postNewTrigger(
         schemaName:String,
         schemaVersion:Int,
         actions:[Dictionary<String, AnyObject>],
         predicate:Predicate,
+        target: Target? = nil,
         completionHandler: (Trigger?, ThingIFError?)-> Void
         )
     {
-        _postNewTrigger(schemaName, schemaVersion: schemaVersion, actions: actions, predicate: predicate, completionHandler: completionHandler)
+        _postNewTrigger(schemaName, schemaVersion: schemaVersion, actions: actions, predicate: predicate, commandTarget: target != nil ? target : self.target, completionHandler: completionHandler)
```

Comment:
since `_postNewTrigger` can accept optional commandTarget, can just call it by:

``` swift
 _postNewTrigger(schemaName, schemaVersion: schemaVersion, actions: actions, predicate: predicate, commandTarget:target, completionHandler: completionHandler)
```